src/Main.elm

```diff
@@ -1,10 +1,198 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
+import Browser exposing (sandbox)
+import Data.Audience exposing (..)
+import Data.AudienceFolder exposing (..)
+import Dict exposing (Dict)
+import Dict.Extra as DE exposing (fromListDedupe)
+import Html exposing (Html, div, button, text)
+import Html.Events
+import Html.Attributes
+import Json.Decode as JD exposing (decodeString)
+import List as L exposing (map, drop, filter)
+import Maybe as M
 
 
-main : Html msg
-main =
-    Html.text "There will be app soon!"
+type alias Model = {
+    audiences : Dict Int (List Audience)
+    , audienceFolders :  Dict Int (List AudienceFolder)
+    , currentFolder : List Int
+    , audienceFilter : AudienceType }
+
+type Msg =
+    Up
+    | Down Int
+    | Filter AudienceType
+
+main : Program () Model Msg
+main = sandbox {
+    init = {audiences = audiences
+        , audienceFolders = audienceFolders
+        , currentFolder = []
+        , audienceFilter = Authored},
+    view = view,
+    update = update}
+
+view : Model -> Html Msg
+view model =
+    div [] [viewContents model]
+
+update : Msg -> Model -> Model
+update msg model =
+    case msg of
+        Up -> {model | currentFolder = L.drop 1 model.currentFolder}
+        Down dir -> {model | currentFolder = dir::model.currentFolder}
+        Filter at -> {model | audienceFilter = at, currentFolder = []}
+
+audiences : Dict Int (List Audience)
+audiences =
+    audiencesJSON
+    |> JD.decodeString decodeAudiences
+    |> Result.withDefault []
+    |> L.map (\aud -> (M.withDefault -1 aud.folder, [aud]))
+    |> DE.fromListDedupe List.append
+
+audienceFolders : Dict Int (List AudienceFolder)
+audienceFolders =
+    audienceFoldersJSON
+    |> JD.decodeString decodeAudienceFolders
+    |> Result.withDefault []
+    |> L.map (\audF -> (M.withDefault -1 audF.parent, [audF]))
+    |> DE.fromListDedupe List.append
+
+viewContents : Model -> Html Msg
+viewContents model =
+    let
+        currF = List.head model.currentFolder |> M.withDefault -1
+        contents =
+            if model.audienceFilter /= Shared then
+                List.append
+                (viewAudienceFolders model.audienceFolders currF)
+                (viewAudiences model.audienceFilter model.audiences currF)
+            else
+                viewAllAudiences
+                <| Dict.foldl (\parent children lst -> children++lst) [] model.audiences
+    in
+    Html.div (contentsStyle (currF /= -1))
+        [ maybeViewBackButton (currF /= -1 && model.audienceFilter /= Shared)
+        , viewFilterButtons model.audienceFilter
+        , Html.ul [Html.Attributes.style "width" "50%"] contents
+        ]
+
+maybeViewBackButton : Bool -> Html Msg
+maybeViewBackButton display =
```

Comment:
Well, every `if` statement is deciding on a bool value. Yes, it is not a `Maybe`, but still it can have only 2 possible outcomes. The function is deciding on a bool value whether to display the button or not.

Reply:
Sorry for badly phrased question. It was about the name of the function implying the wrong input type. As the function name starts with `maybe` I expect `Maybe` to be decided on.

I would change the name, not the inner `if` nor `Bool` as an argument. Something like [viewIf](https://package.elm-lang.org/packages/elm-community/html-extra/3.4.0/Html-Extra#viewIf)

Reply:
I think I understood your question. But I did not know, there is a convention. That when a function is deciding on `Maybe`, the function's name should start with `maybe...`. And when deciding on bool, the function's name should end with `If`.